File: seminar/src/main/java/com/server/dosopt/seminar/repository/ServiceMemberRepository.java
package com.server.dosopt.seminar.repository;

import com.server.dosopt.seminar.domain.ServiceMember;
import java.util.Optional;
import org.springframework.data.jpa.repository.JpaRepository;

public interface ServiceMemberRepository extends JpaRepository<ServiceMember, Long> {

    Optional<ServiceMember> findByNickname(String nickname);
}

File: kotlin-seminar/src/main/kotlin/sopt/org/kotlinseminar/repository/ServiceMemberRepository.kt
package sopt.org.kotlinseminar.repository

import org.springframework.data.jpa.repository.JpaRepository
import sopt.org.kotlinseminar.domain.ServiceMember
import java.util.*


interface ServiceMemberRepository: JpaRepository<ServiceMember, Long> {
    fun findByNickname(nickname: String): Optional<ServiceMember>
}